### PR TITLE
Add srcarch, nvrs and nevrs extension tags to return src as "architecture" for srpms

### PR DESCRIPTION
**Pull request by ffesti**
Based on #1116 Should probably be squashed with the original patch.

This make the original more feature complete and does not limit the change to rpmspec but makes thie new behaviour default for all queries.

### CODE REVIEW

**Comment by pmatilai** [comment]
A testcase or two for the new tags, plus doc update missing.

File: lib/tagexts.c [hunk 1]
```diff
@@ -776,6 +777,17 @@ static int getNEVRA(Header h, rpmtd td, nevraFlags flags)
 	if (headerIsSource(h) && val == NULL) val = "src";
 	if (val) rstrscat(&res, ".", val, NULL);
     }
+    if ((flags & NEVRA_SRCARCH)) {
+	if (headerIsSource(h)) {
+	    if (headerIsEntry(h, RPMTAG_NOSOURCE))
```
Comment (mlschroe) [inline lib/tagexts.c:782]: You also need to check for RPMTAG_NOPATCH

**Comment by mlschroe** [comment]
Changing the way the old queries work is considered too dangerous?

**Comment by pmatilai** [comment]
Yeah we discussed the options and I'm pretty torn on this. Adding new tags for this seems redundant and silly especially since RPMTAG_NEVRA can already return .src as the arch in special cases, but it's also a safe route.

**Comment by Conan-Kudo** [comment]
> Yeah we discussed the options and I'm pretty torn on this. Adding new tags for this seems redundant and silly especially since RPMTAG_NEVRA can already return .src as the arch in special cases, but it's also a safe route.

I dunno, I think I'd rather change the existing things than adding new stuff. This stuff is not free, because everything else has to adapt to explicitly use the new things. ☹️

**Comment by ffesti** [comment]
One aspect is that it is very awkward to offer this behaviour for NEVRA only and not have a tag that offers the same information for the arch only. But as SRPMs are arch dependent, we can't just overwrite the behaviour of ARCH (although we briefly considered having an extension tag with the same name so one could get back tot the old ebhaviour by switching extension tags off). If we can come up with a new name that starts with A changing the behaviour of NEVRA becomes more mamagable and less ugly. May be ARCHSRC?

**Comment by pmatilai** [comment]
Oh, I think were talking about the NVRS tags being redundant. The special arch handling does deserve a separate new extension tag, ie the RPMTAG_SRCARCH here, and then NEVRA is changed to use that.

**Comment by ffesti** [comment]
Yes, that's what I am talking about, too. Have one new ARCH-like tag and change NEVRA (and NVRA) to use that. It would be nice if the tag start with an A so the A in NEVRA still points to some tag in a meaningful way. NEVRA meaning Name, Epoch, Version, Release, srcArch only adds more confusion.

**Comment by pmatilai** [comment]
Yup. Both SRCARCH and ARCHSRC are bad names because they suggest that this is relevant only for source packages when that's not the case. DISPLAYARCH or something to that effect may be closer.

**Comment by ffesti** [comment]
What about ARCHSUFFIX? The we could just change NEVRA to use that.

**Comment by ffesti** [comment]
OK, I made a branch with an ARCHSUFFIX extension tag that is then used in NEVRA and friends. Results are failures in the test suite looking like:

```
error: Failed build dependencies:
-       foo > 1.2.3 is needed by deptest-buildreq-1.0-1.noarch
+       foo > 1.2.3 is needed by deptest-buildreq-1.0-1.src
```
Not 100% sure how I feel about that. On one hand - yes, this is an issue building ans source package and may be the main package is not even going to be built. Otoh having the arch there makes the error message more useful.

**Comment by pmatilai** [comment]
Okay that's useful (trying things out over speculating often is): as build dependencies can and will vary across architectures, the real arch *has to* be present in the build dependency output for diagnostics. So that puts a nice and firm end to the speculation whether we can just change NEVRA or not - we can't.

ARCHSUFFIX is kinda odd, but it makes sense if you know to associate it with a file name.

**Comment by ffesti** [comment]
Added check for NOPATCH.

**Comment by ffesti** [comment]
OK, if we can't change NEVRA we need to add some new tags. ARCHSUFFIX is a better name than SRCARCH and ARCHSRC.
May be we can just change the default for queries to NEVR.ARCHSUFFIX and get rid of all this new NEVRS character soup.

**Pull request closed without merging**